ngraph/core/include/openvino/core/descriptor/tensor.hpp

```diff
@@ -97,6 +98,8 @@ class OPENVINO_API Tensor {
     ngraph::HostTensorPtr m_lower_value, m_upper_value;
     std::string m_name;
     std::unordered_set<std::string> m_names;
+    size_t m_instance_id{m_next_instance_id.fetch_add(1)};
+    static std::atomic<size_t> m_next_instance_id;
```

Comment:
what if `m_instance_id` is generated once `get_names` is called the first time e.i. `m_names` are empty? In this case once we set `set_names({})` and then `get_names()` -- new ID will be generated